components/ReviewCard.tsx

```diff
@@ -106,6 +111,110 @@ export const ReviewCard: React.FC<ReviewCardProps> = ({ submissionData }) => {
       }
     }
   }
+  return (
+    <>
+      <div className="d-flex justify-content-between px-2 py-1">
+        <label>
+          <input
+            type="radio"
+            checked={commentType === 'accept'}
+            value="accept"
+            onChange={onChange}
+            className="mr-2"
+          />
+          <p className="font-weight-bold d-inline">Accept</p>
+          <p className="text-muted">Submit feedback and approve submission</p>
+        </label>
+        <label>
+          <input
+            type="radio"
+            checked={commentType === 'reject'}
+            value="reject"
+            onChange={onChange}
+            className="mr-2"
+          />
+          <p className="font-weight-bold d-inline">Reject</p>
+          <p className="text-muted">Request changes and reject submission</p>
+        </label>
+        <label>
+          <input
+            type="radio"
+            checked={commentType === 'comment'}
+            value="comment"
+            onChange={onChange}
+            className="mr-2"
+          />
+          <p className="font-weight-bold d-inline">Comment</p>
+          <p className="text-muted">
+            Submit general feedback without explicit approval
+          </p>
+        </label>
+      </div>
+      <Button type="success" color="white" onClick={() => submitReview()}>
+        Submit
+      </Button>
+    </>
+  )
+}
+
+export const ReviewCard: React.FC<ReviewCardProps> = ({ submissionData }) => {
+  const {
+    id,
+    diff,
+    updatedAt,
+    user,
+    challenge,
+    lessonId,
+    reviewer,
+    challengeId,
+    status
+  } = submissionData
+  const context = useContext(GlobalContext)
+  const name = context.session?.user?.name
+  const username = context.session?.user?.username
+  if (!name || !username) {
+    return (
+      <Error
+        code={StatusCode.INTERNAL_SERVER_ERROR}
+        message={'Error while retrieving userinfo from context'}
+      />
+    )
+  }
+  const [index, setIndex] = useState(-1)
+  const [commentValue, setCommentValue] = useState('')
+  const [commentType, setCommentType] = useState<CommentType>('comment')
+  const [submissionState, setSubmission] = useState<Submission>(submissionData)
+  const { data, loading, error } = useGetPreviousSubmissionsQuery({
+    variables: { challengeId, userId: user.id }
+  })
+  const [previousSubmissions, setPreviousSubmissions] = useState(data)
+  const update = updateCache(
+    submissionState.id,
+    commentValue,
+    name,
+    username,
+    lessonId,
+    undefined,
+    undefined,
+    challengeId,
+    user.id
+  )
+  const underComments = submissionState.comments?.filter(
+    comment => !comment.line
+  )
+  useEffect(() => {
+    if (data?.getPreviousSubmissions) {
+      setPreviousSubmissions(data)
+      /*istanbul ignore else*/
```

Comment:
I spent quite a lot of time trying to trigger this else case but to no avail. I couldn't get the final rerender from updating the cache in tests no matter what.

Reply:
I've managed to finally trigger that branch by hardcoding comment for every cache update. I guess it's better than completely ignoring it.